dotCMS/src/main/java/com/dotmarketing/quartz/job/IntegrityDataGenerationJob.java

```diff
@@ -0,0 +1,146 @@
+import com.dotmarketing.quartz.DotStatefulJob;
+import com.dotmarketing.quartz.QuartzUtils;
+import com.dotmarketing.util.Logger;
+import com.rainerhahnekamp.sneakythrow.Sneaky;
+import org.quartz.InterruptableJob;
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.SimpleTrigger;
+import org.quartz.UnableToInterruptJobException;
+
+import java.util.Date;
+
+/**
+ * Quartz stateful job in charge of generating the integrity data for a provided endpoint id and request id.
+ * It calls in place functionality at {link IntegrityUtil} to do the actual generation of the zip file.
+ *
+ * It also implements the {@link InterruptableJob} interface to support cancellation at any time.
+ */
+public class IntegrityDataGenerationJob extends DotStatefulJob implements InterruptableJob {
+
+    private static final String TRIGGER_NAME = "IntegrityDataGenerationTrigger";
+    private static final String TRIGGER_GROUP = "integrity_data_generation_triggers";
+
+    public static final String JOB_NAME = "IntegrityDataGenerationJob";
+    public static final String JOB_GROUP_NAME = "integrity_data_generation_jobs";
+
+    private JobExecutionContext jobContext;
+
+    /**
+     * Code to execute when Quartz calls the job.
+     * Takes endpoint and request id from context to tries to generate check data zip file not without adding some
+     * metadata to have control of what is going on.
+     *
+     * @param jobContext job context
+     * @throws JobExecutionException
+     */
+    @Override
+    @CloseDBIfOpened
+    public void run(final JobExecutionContext jobContext) throws JobExecutionException {
+        this.jobContext = jobContext;
+        final JobDataMap jobDataMap = this.jobContext.getJobDetail().getJobDataMap();
+        final PublishingEndPoint requesterEndPoint =
+                (PublishingEndPoint) jobDataMap.get(IntegrityUtil.REQUESTER_ENDPOINT);
+        final String requestId = (String) jobDataMap.get(IntegrityUtil.INTEGRITY_DATA_REQUEST_ID);
+
+        IntegrityUtil.saveIntegrityDataStatus(
+                requesterEndPoint.getId(),
+                requestId,
+                IntegrityResource.ProcessStatus.PROCESSING);
+
+        try {
+            // Actual integrity data file generation
+            IntegrityUtil.generateDataToCheckZip(requesterEndPoint.getId());
+            // Integrity data generation went ok
+            IntegrityUtil.saveIntegrityDataStatus(
+                    requesterEndPoint.getId(),
+                    requestId,
+                    IntegrityResource.ProcessStatus.FINISHED);
+            Logger.info(
+                    IntegrityDataGenerationJob.class,
+                    String.format("Job execution for endpoint %s has finished", requesterEndPoint.getId()));
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6151243)